src/storage/src/hummock/store/version.rs

```diff
@@ -537,22 +537,29 @@ pub fn read_filter_for_batch(
 
     let mut imm_vec = Vec::default();
     let mut sst_vec = Vec::default();
+    let mut seen_imm_ids = HashSet::new();
+    let mut seen_sst_ids = HashSet::new();
 
     // only filter the staging data that epoch greater than max_mce to avoid data duplication
     let (min_epoch, max_epoch) = (max_mce_version.max_committed_epoch(), epoch);
     // prune imm and sst with max_mce
     for (staging_imms, staging_ssts) in staging_vec {
-        imm_vec.extend(
-            staging_imms
-                .into_iter()
-                .filter(|imm| imm.min_epoch() > min_epoch && imm.min_epoch() <= max_epoch),
-        );
+        imm_vec.extend(staging_imms.into_iter().filter(|imm| {
+            // There shouldn't be duplicated IMMs because merge imm only operates on a single shard.
```

Comment:
Previously I thought merge imm will merge the data of all shards because I saw the parameter `instance_id` of `build_shared_buffer_batch` is an option, and I thought the reason for this is that merge imm will merge the imms from multiple shards🤣 But it seems that only in testing we will have `build_shared_buffer_batch` with None instance_id. Can we by the way in this PR to change the None `instance_id` parameter of `build_shared_buffer_batch` to an explicit instance id?